providers/tentacle.rb

```diff
@@ -172,6 +172,16 @@
     end
     new_resource.updated_by_last_action(actions_updated?([register_polling_instance]))
   else
+
+    firewall = windows_firewall_rule 'Octopus Deploy Tentacle' do
+      localport port.to_s
+      dir :in
+      protocol 'TCP'
+      firewall_action :allow
+      action :create
```

Comment:
Since the firewall only applies to the listening tentacles, it is at the top of the else statement. 

It might be slightly prettier code if the logic was flipped so that listening tentacles are first in the if statement and listening are part of the 'else'